contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,319 @@
+    function _getExpiredPutSpreadCashValue(
+        FixedPointInt256.FixedPointInt memory _shortAmount,
+        FixedPointInt256.FixedPointInt memory _longAmount,
+        FixedPointInt256.FixedPointInt memory _shortCashValue,
+        FixedPointInt256.FixedPointInt memory _longCashValue
+    ) internal pure returns (FixedPointInt256.FixedPointInt memory) {
+        return _shortCashValue.mul(_shortAmount).sub(_longCashValue.mul(_longAmount));
+    }
+
+    /**
+     * @dev calculate cash value for an expired call spread vault.
+     *                     (short cash value * short amount) - ( long cash value * long Amount )
+     *  Formula: net =   -------------------------------------------------------------------------
+     *                                               Underlying price
+     * @return net value
+     */
+    function _getExpiredCallSpreadCashValue(
+        FixedPointInt256.FixedPointInt memory _shortAmount,
+        FixedPointInt256.FixedPointInt memory _longAmount,
+        FixedPointInt256.FixedPointInt memory _shortCashValue,
+        FixedPointInt256.FixedPointInt memory _longCashValue,
+        FixedPointInt256.FixedPointInt memory _underlyingPriceInt
+    ) internal pure returns (FixedPointInt256.FixedPointInt memory) {
+        return _shortCashValue.mul(_shortAmount).sub((_longCashValue.mul(_longAmount))).div(_underlyingPriceInt);
+    }
+
+    /**
+     * @dev ensure that the vault contains
+     * a) at most 1 asset type used as collateral,
+     * b) at most 1 series of option used as the long option and
+     * c) at most 1 series of option used as the short option.
+     * @param _vault the vault to check.
+     */
+    function _checkIsValidSpread(MarginAccount.Vault memory _vault) internal pure {
+        require(_vault.shortOtokens.length <= 1, "MarginCalculator: Too many short otokens in the vault.");
+        require(_vault.longOtokens.length <= 1, "MarginCalculator: Too many long otokens in the vault.");
+        require(_vault.collateralAssets.length <= 1, "MarginCalculator: Too many collateral assets in the vault.");
+
+        require(
+            _vault.shortOtokens.length == _vault.shortAmounts.length,
+            "MarginCalculator: Short asset and amount mismatch."
+        );
+        require(
+            _vault.longOtokens.length == _vault.longAmounts.length,
+            "MarginCalculator: Long asset and amount mismatch."
+        );
+        require(
+            _vault.collateralAssets.length == _vault.collateralAmounts.length,
+            "MarginCalculator: Collateral asset and amount mismatch."
+        );
+    }
+
+    /**
+     * @dev if there is a short option in the vault, ensure that the long option series being used is a valid margin.
+     * @param _vault the vault to check.
+     */
+    function _isMarginableLong(MarginAccount.Vault memory _vault) internal view returns (bool) {
+        if (_vault.longOtokens.length == 0 || _vault.shortOtokens.length == 0) return true;
```

Comment:
this one also should be fixed
`if(
  (_vault.longOtokens.length == 0 || _vault.longOtokens[0] == address(0)) ||
  (_vault. shortOtokens.length == 0 || _vault. shortOtokens[0] == address(0))
) {
 return true
}